dotcom-rendering/src/components/ShowHideContainers.importable.tsx

```diff
@@ -53,15 +73,22 @@ export const ShowHideContainers = () => {
 
 		for (const e of allShowHideButtons) {
 			const sectionId = e.getAttribute('data-show-hide-button');
+			const isBetaContainer = e.getAttribute('data-beta-container');
 			if (!sectionId) continue;
 
-			e.onclick = () => toggleContainer(sectionId, e);
+			if (isSignedIn === false && isBetaContainer === 'true') {
```

Comment:
Following on from my [other comment](https://github.com/guardian/dotcom-rendering/pull/13176#issuecomment-2607193951), if we are later going to remove the `isBetaContainer` check, I would be tempted to separate out the logic for what happens when the container is a beta container from when it isn't, i.e. favouring duplication to make it easier to remove the unnecessary code later on.

i.e.
```
if (isBetaContainer === 'true') {
...
} else {
...
}
```

Reply:
I have reworked this logic so (hopefully) its easier to understand. 

In the immediate, the behaviour is as follows: 
 - If the container is a beta container, we only allow show hide if the user is signed in
 - If the container is a existing / legacy container, we always allow show hide.